From: Arnaud Ferraris <aferraris@debian.org>
Date: Tue, 5 Nov 2024 16:45:35 +0100
Subject: Update xkbcommon dependency to 0.8

This is the latest upstream version and it fixes a bug in the handling
of UTF-8 characters.

Forwarded: https://gitlab.gnome.org/World/Phosh/squeekboard/-/merge_requests/697
---
 Cargo.toml          | 2 +-
 src/data/parsing.rs | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 0851d5c..dee8cdc 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,7 +34,7 @@ clap = { version = "4.4", features=["std"], default-features = false }
 zbus = "1.9"
 zvariant = "2.10"
 zvariant_derive = "2.10"
-xkbcommon = { version = "0.7", features = ["wayland"] }
+xkbcommon = { version = "0.8", features = ["wayland"] }
 
 [dependencies.cairo-rs]
 version = "0.18"
diff --git a/src/data/parsing.rs b/src/data/parsing.rs
index 7c8236f..6aaf78f 100644
--- a/src/data/parsing.rs
+++ b/src/data/parsing.rs
@@ -696,7 +696,7 @@ mod tests {
             xkb::KEYSYM_NO_FLAGS,
         );
         let keysym = xkb::keysym_to_utf8(keysym);
-        assert_eq!(keysym, "å\0");
+        assert_eq!(keysym, "å");
     }
     
     #[test]
